File: backend/src/main/java/com/xxgwy/modules/user/repository/UserRepository.java
package com.xxgwy.modules.user.repository;

import com.xxgwy.modules.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface UserRepository extends JpaRepository<User, UUID> {
    Optional<User> findByUsername(String username);
    Optional<User> findByOpenid(String openid);
    boolean existsByUsername(String username);
}

File: backend/src/main/kotlin/com/xxgwy/modules/user/repository/UserRepository.kt
package com.xxgwy.modules.user.repository

import com.xxgwy.modules.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface UserRepository : JpaRepository<User, UUID> {
    fun findByUsername(username: String): Optional<User>
    fun existsByUsername(username: String): Boolean
}
